blockservice/blockservice.go

```diff
@@ -133,23 +157,13 @@ func (s *blockService) Allowlist() verifcid.Allowlist {
 	return s.allowlist
 }
 
-// NewSession creates a new session that allows for
-// controlled exchange of wantlists to decrease the bandwidth overhead.
-// If the current exchange is a SessionExchange, a new exchange
-// session will be created. Otherwise, the current exchange will be used
-// directly.
-// Sessions are lazily setup, this is cheap.
-func NewSession(ctx context.Context, bs BlockService) *Session {
-	ses := grabSessionFromContext(ctx, bs)
-	if ses != nil {
-		return ses
-	}
-
-	return newSession(ctx, bs)
```

Comment:
I'm not sure why session-grabbing from context is removed... I remember old discussions about embedding sessions in contexts etc. but not sure if used...